### PR TITLE
fix(components): [space] missing empty string size

### PR DESCRIPTION
**Pull request by HeftyKoo**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by jw-foss** [comment]
Syntactically I don't think we need this maybe add fallbacks to `falsy` value might be a better solution?

**Comment by tolking** [comment]
This is indeed a `ts` syntax error, maybe the components size should not accept `''` value.

**Comment by HeftyKoo** [comment]
> Syntactically I don't think we need this maybe adding fallbacks to `falsy` value might be a better solution?

It's more than a typo issue, may be exclude the empty string is better.

File: packages/components/space/src/use-space.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import type { SpaceProps } from './space'
 import type { CSSProperties, StyleValue } from 'vue'
 import type { ComponentSize } from '@element-plus/constants'
 
-const SIZE_MAP: Record<ComponentSize, number> = {
+const SIZE_MAP: Record<Exclude<ComponentSize, ''>, number> = {
```
Comment (tolking) [inline packages/components/space/src/use-space.ts:9]: There mark a type error on line 60, and you should also modify line 60

**Review by tolking — CHANGES_REQUESTED** [review]

**Review by tolking — APPROVED** [review]

File: packages/components/space/src/use-space.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export function useSpace(props: SpaceProps) {
       if (isNumber(size)) {
         val = size
       } else {
-        val = SIZE_MAP[size] || SIZE_MAP.small
+        val = SIZE_MAP[size || 'small'] || SIZE_MAP.small
```
Comment (tolking) [inline packages/components/space/src/use-space.ts:60]: It seems that both are the same

```diff
- val = SIZE_MAP[size || 'small'] || SIZE_MAP.small
+ val = SIZE_MAP[size || 'small']
```

File: packages/components/space/src/use-space.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import type { SpaceProps } from './space'
 import type { CSSProperties, StyleValue } from 'vue'
 import type { ComponentSize } from '@element-plus/constants'
 
-const SIZE_MAP: Record<ComponentSize, number> = {
+const SIZE_MAP: Record<Exclude<ComponentSize, ''>, number> = {
   small: 8,
   default: 12,
   large: 16,
```
Comment (sxzz) [inline packages/components/space/src/use-space.ts:12]:
```suggestion
const SIZE_MAP = {
  small: 8,
  default: 12,
  large: 16,
) as const
```

File: packages/components/space/src/use-space.ts (hunk 2 above)
Comment (sxzz) [inline packages/components/space/src/use-space.ts:60]:
```suggestion
        val = SIZE_MAP[size || 'small']
```

**Review by sxzz — APPROVED** [review]

**Review by tolking — APPROVED** [review]

File: packages/components/space/src/use-space.ts (hunk 2 above)
Comment (sxzz) [reply packages/components/space/src/use-space.ts:59]: if `size` is an invalid value, it will fallback to `small`

**Pull request merged**